pilot/pkg/model/status/helper.go

```diff
@@ -70,25 +71,20 @@ func UpdateConfigCondition(cfg config.Config, condition *v1alpha1.IstioCondition
 		cfg.Status = &v1alpha1.IstioStatus{}
 	}
 	status = cfg.Status.(*v1alpha1.IstioStatus)
-	status.Conditions = UpdateCondition(status.Conditions, condition)
+	status.Conditions = updateCondition(status.Conditions, condition)
 	return cfg
 }
 
-func UpdateCondition(conditions []*v1alpha1.IstioCondition, condition *v1alpha1.IstioCondition) []*v1alpha1.IstioCondition {
+func updateCondition(conditions []*v1alpha1.IstioCondition, condition *v1alpha1.IstioCondition) []*v1alpha1.IstioCondition {
 	ret := append([]*v1alpha1.IstioCondition(nil), conditions...)
```

Comment:
we donot need to copy conditions here, increasing another mem allocation

Reply:
As with my previous idea, we can update or delete without a copy condition, thanks